mlkem768_test.go

```diff
@@ -79,6 +80,72 @@ func TestDecompressCompress(t *testing.T) {
 	}
 }
 
+func CompressRat(x, d int64) int64 {
+	if (x < 0) || (x >= q) {
+		panic("x out of range")
+	}
+	if (d <= 0) || (d >= 12) {
+		panic("d out of range")
+	}
+
+	precise := big.NewRat((1<<d)*x, q) // (2ᵈ / q) * x == (2ᵈ * x) / q
+
+	// FloatString rounds halves away from 0, and our result should always be positive,
+	// so it should work as we expect.
+	// (There's no direct way to convert a Rat to an integer)
+	rounded, err := strconv.ParseInt(precise.FloatString(0), 10, 64)
+	if err != nil {
+		panic(err)
+	}
+
+	return rounded % (1 << d)
```

Comment:
Why the modulo? I thought the result should always come out to less than 2ᵈ since x < q.

Reply:
(Similarly below.)

Reply:
Consider d=1, x=q-1

`(2ᵈ / q) * x` gives `(2/q)*(q-1)`, which evaluates to `1.99...`, rounding up to 2 - which is outside the range of [0..2ᵈ), requiring the reduction

Reply:
I'll add a comment noting this